### PR TITLE
Don't Merge! Question

### PR DESCRIPTION
**Pull request by rcoletti116**
Hi @jimhe I have been attempting to extend the plugin based on your platform architecture. I must be missing something pretty fundamental with how you've written the plugin. 

If you have time would you mind looking at my fork? The PR here attempts to add an additional accessory leveraging Infinitude's OAT value for outdoor temperature. No matter what I do I cannot get the platform to add the new accessory.

I updated InfinitudePlatform.js and created a new file called InfinitudeSensor.js.

I'll take any pointers since you wrote the plugin.

### CODE REVIEW

File: src/InfinitudePlatform.js [hunk 1]
```diff
@@ -3,8 +3,9 @@ const { pluginName, platformName } = require('./constants');
 const configSchema = require('./configSchema');
 const InfinitudeClient = require('./InfinitudeClient');
 const InfinitudeThermostat = require('./InfinitudeThermostat');
+const InfinitudeSensor = require('./InfinitudeSensor');
 
-let AccessoryCategories, Thermostat;
+let AccessoryCategories, Thermostat,Sensor;
```
Comment (jimhe) [inline src/InfinitudePlatform.js]: `Sensor` should be `TemperatureSensor`

File: src/InfinitudePlatform.js [hunk 2]
```diff
@@ -43,6 +49,15 @@ module.exports = class InfinitudePlatform {
...
   }
+	
+ configureAccessory(accessory) {
+    this.initializeZones(false).then(
+      function() {
+        this.accessories[accessory.UUID] = accessory;
+        this.configureSensorAccessory(accessory);
+      }.bind(this)
+    );
+  }
```
Comment (jimhe) [inline src/InfinitudePlatform.js]: I don't think you can dupliate `configureAccessory` like this. You'll have to alter the function that's already there.

**Comment by jimhe** [comment]
@rcoletti116 here is an example of how you might go about creating a new accessory: https://github.com/jimhe/homebridge-infinitude/commit/49f85bf24ad814e61750b0e8d84b8f156e88c3b4

Unfortunately the code as written is very specific to zones so the example commit is rather hacky. But the idea is that we create the new sensor accessory in `initializeZones` and configure it if we are looking at `OutsideUuid` which points to it.

I haven't tested the code so there are no guarantees that it'll work but hopefully you understand what I was trying to do in it.

**Comment by rcoletti116** [comment]
Thanks for looking! I've gotten so far as to get the sensor to show up in HomeKit, but have not got it to display the correct value. Also, it seems to have an either/or effect for the sensor and thermostat, and no thermostats are created. Both could be related to a TypeError i've been troubleshooting.

`node:17053) UnhandledPromiseRejectionWarning: TypeError: platformAccessory.getService is not a function
    at new InfinitudeSensor (/usr/local/lib/node_modules/homebridge-infinitude/src/InfinitudeSensor.js:13:39)
    at InfinitudePlatform.configureSensorAccessory (/usr/local/lib/node_modules/homebridge-infinitude/src/InfinitudePlatform.js:133:5)
    at InfinitudePlatform.<anonymous> (/usr/local/lib/node_modules/homebridge-infinitude/src/InfinitudePlatform.js:50:16)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)

(node:17053) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)

**Pull request closed without merging**